.github/workflows/ci.yaml

```diff
@@ -1,8 +1,6 @@
 name: Create-and-publish-a-Docker-image
 
-on:
-  push:
-    branches: [ master ]
+on: push
```

Comment:
so the problem is this:
any branch commit will result in this action triggered. feature/branch that developer is using for current development will push changes to the environment - which is 100% risky and buggy.

you should limit what branches triggers the build.

Also what I meant in my previous feedback is, sometimes you need to test a new version of the pipeline, but you don't need/want to break the current production pipeline.

You start a new branch `feature/ci-improvements` and perform the changes there, while main branch still has your OLD implementation (that is supposed to be working 100% of the time). Then once you are comfortable with your changes - your merge `feature/ci-improvements -> master` and your master has a new pipeline implementation and you have 0% downtime.

So test this scenario and limit the branches you can trigger the github action

Reply:
I realized that the build on each branch makes its own package with the appropriate tag. Thus, packages for the production environment, as well as for testing and development, can be stored in the repository. Depending on the branch, we will specify different parameters in ci.yaml:
for master:
on:
  push:
    branches: [ master ]

for dev:
on:
  push:
    branches: [ dev ]
for feature/ci-improvements:
on:
  push:
    branches: [ feature/ci-improvements]